code/modules/mob/living/living_status_procs.dm

```diff
@@ -721,6 +721,7 @@ STATUS EFFECTS
 		return
 	var/datum/status_effect/incapacitating/floored/F = IsKnockedDown()
 	if(amount <= 0)
+		remove_status_effect(STATUS_EFFECT_BATON_KNOCKDOWN) // we want to cancel delayed baton knockdowns as well
```

Comment:
code looks good, I don't know stun code great so it would be nice to hear feedback on whether or not this is scoped to the right location

Reply:
I feel like this should be done using a signal. 
send a signal on `SetKnockdown` then have the status effect listen for it and when it hears it remove the status effect.

Reply:
will do when I get a chance